### ISSUE TITLE
Fleet takes a long time to come up on GKE HA local cluster; initial fleet install tries with  bad RC `fleet-100.0.0-up0.3.6- rc11` for the first 10 minutes, then switches to rc12

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#34265]
**Rancher Server Setup**
- Rancher version: v2.6.0-rc3
- Installation option (Docker install/Helm Chart): helm
   - If Helm Chart, Kubernetes Info: v1.20.8-gke900
         Cluster Type (RKE1, RKE2, k3s, EKS, etc): GKE
         Node Setup: 3 nodes, e2 medium
         Version:1.20.8-gke.900
- Proxy/Cert Details: n/a

**Information about the Cluster**
- Kubernetes version:1.20.8-gke.900
- Cluster Type (Local/Downstream):local
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**Describe the bug**

fleet took about 15-20 minutes to come to a good state. There were about 12 failed `helm-operations` jobs related to the fleet install for each RC. Error messages for the first ~10 minutes show that rancher is trying to use rc11 (a bad version of fleet). After this period, rancher then uses the latest rc (rc12) and is able to finally come up successfully (after about the same number of retries by `helm-operations` pods)
```
helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.0-up0.3.6-rc11.yaml --version=100.0.0+up0.3.6-rc11 --wait=true fleet /home/shell/helm/fleet-100.0.0-up0.3.6-rc11.tgz
```

then, after ~10 more minutes
```
SUCCESS: helm upgrade --force-adopt=true --history-max=5 --install=true --namespace=cattle-fleet-system --reset-values=true --timeout=5m0s --values=/home/shell/helm/values-fleet-100.0.0-up0.3.6-rc12.yaml --version=100.0.0+up0.3.6-rc12 --wait=true fleet /home/shell/helm/fleet-100.0.0-up0.3.6-rc12.tgz
```

**To Reproduce**
 
* provision rancher v2.6.0-rc3 running on [GKE cluster](https://rancher.com/docs/rancher/v2.5/en/installation/install-rancher-on-k8s/gke/)
* after rancher is running, visit the cluster through your browser, and navigate to `local` -> `workloads` -> pods

**Result**
* looking at the fleet-related `helm-operations` pods, fleet is attempting to install the wrong version. 

**Expected Result**

rancher should use the latest rc from fleet

**Screenshots**

**Additional context**

**Comment by nickgerace** [comment]
I do not see this issue on `v2.6-head`. Since rc11 and rc12 were tagged around Rancher 2.6.0-rc3, we should test again on the next Rancher RC.

**Comment by brendarearden** [comment]
Available for testing in 2.6.0-rc4

**Comment by slickwarren** [comment]
this is fixed on 2.6.0-rc4, rancher pod is pulling the latest RC from fleet and fleet is able to come up successfully in an airgap setup.
